model/src/main/java/org/projectnessie/model/SqlView.java

```diff
@@ -27,17 +29,16 @@
 @JsonTypeName("VIEW")
 public abstract class SqlView extends Contents {
 
-  public static enum Dialect {
-    HIVE,
-    SPARK,
-    DREMIO,
-    PRESTO
-  }
-
   public abstract String getSqlText();
 
+  @Nullable
+  public abstract String getResolvedSqlText();
+
+  @Nullable
+  public abstract String getSqlContext();
+
   public abstract Dialect getDialect();
 
-  // Schema getSchema();
+  public abstract Schema getSchema();
```

Comment:
The client has to send the `Schema` in Nessies own format. At this time I don't plan on any engine specific extensions. I plan on adding 2 new modules to Nessie: 1 for Iceberg -> Nessie and 1 for Arrow -> Nessie, these will be kept in sync but its up to other engines to maintain sync w/ their internal formats.

I would like our `Schema` to be a **thin** wrapper around a well known format eg Iceberg or Arrow. The only purpose to defining our own containers is to avoid extra dependencies for the client.